pipeline/rds_to_s3.py

```diff
@@ -0,0 +1,125 @@
+"""
+Contains code to retrieve watering/recording data older than 24hrs from the database, append it to
+the relevant files in s3 (automatically creates if none exist), and delete from the db.
+"""
+
+from os import environ
+from datetime import datetime, timedelta
+import pandas as pd
+from dotenv import load_dotenv
+import sqlalchemy as db
+from sqlalchemy.engine.base import Connection
+from boto3 import client
+
+
+load_dotenv()
+STR_DAY_AGO = (datetime.now() - timedelta(hours=24)).strftime("%Y-%m-%d %H:%M:%S")
+TODAY = datetime.today()
+
+
+def get_database_engine():
+    """Returns the database engine."""
+    try:
+        engine = db.create_engine(
+            f"mssql+pymssql://{environ['DB_USER']}:{environ['DB_PASSWORD']}@{environ['DB_HOST']}:{environ['DB_PORT']}/{environ['DB_NAME']}?charset=utf8"
+        )
+        return engine
+    except Exception as e:
+        print(f"Error creating database engine: {e}")
+        raise e
+
+
+def get_old_records(table_name: str, db_engine: db.Engine, connection: Connection, metadata,
+                    date_cutoff: str = STR_DAY_AGO):
+    """
+    Retrieves records older than 24 hours (by attribute 'datetime') from db table with given name
+    (watering/recording).
+    """
+    try:
+        table = db.Table(table_name, metadata, autoload_with=db_engine)
+        query = db.select(table).where(table.columns.datetime < str.encode(date_cutoff))
+        response = connection.execute(query)
+        return pd.DataFrame(response.fetchall())
+
+    except Exception as e:
+        raise e
+
+
+def get_day_bucket_keys(s3_client: client, folder_path: str, day: int = TODAY.day,
+                    bucket_name: str = environ['BUCKET_NAME']) -> list:
+    """
+    Returns list of keys within a given s3 bucket ending in '_{day}.csv, with a prefix matching
+    the given folder_path.
+    """
+    objects = s3_client.list_objects(Bucket=bucket_name, Prefix=folder_path).get('Contents')
+    if objects:
+        return [obj['Key'] for obj in objects if obj['Key'].split('_')[-1] == f'{day}.csv']
+    return []
+
+
+def get_current_csv_data(data_type: str, s3_client: client, bucket_name:
+                         str = environ['BUCKET_NAME'], date: str = TODAY) -> pd.DataFrame:
+    """
+    Downloads relevant files of specified data_type (watering/recording) from S3 to local, and
+    then returns it as a pandas dataframe
```

Comment:
Needs a full stop at end of doc string to be consistent with others

Reply:
Very true, good spot